### PR TITLE
doc: add reference to RIOT paper

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
This adds a reference to the newest [RIOT paper](https://riot-os.org/docs/riot-ieeeiotjournal-2018.pdf) to documentation.

### Testing procedure
`make doc` and open `doc/doxygen/html/index.html`. The amended text should now appear.

### Issues/PRs references
None

### CODE REVIEW

**Comment by miri64** [comment]
> Do you have access to the published pdf version ?

The published version can be found on the [IEEE website](https://ieeexplore.ieee.org/abstract/document/8315125). However, you have to be an IEEE subscriber (which I am not) to be able to read it.

**Comment by cladmi** [comment]
If I understood properly how published paper works:
Usually you must pay to access the published version and are not allowed to freely re-distribute it because of copyright.
But you can give the author/"pre editor" version as you wish.

**Comment by aabadie** [comment]
Maybe the DOI link could also be added ?

**Comment by miri64** [comment]
BTW, DOI is provided in the PDF's header ;-).

File: doc/doxygen/src/mainpage.md [hunk 1]
```diff
@@ -23,6 +23,10 @@ Across all supported hardware (32-bit, 16-bit, and 8-bit platforms). RIOT
 provides a consistent API and enables ANSI C and C++ application programming,
 with  multithreading, IPC, system timers, mutexes etc.
 
+[This article](https://riot-os.org/docs/riot-ieeeiotjournal-2018.pdf), published
+in the IEEE Internet of Things Journal, volume 5 issue 6, provides a good
```
Comment (emmanuelsearch) [inline doc/doxygen/src/mainpage.md]: how about:
A good high-level overview can be found in the article [RIOT: An Open Source Operating System for Low-End Embedded Devices in the IoT](https://riot-os.org/docs/riot-ieeeiotjournal-2018.pdf) (IEEE Internet of Things Journal, December 2018).

Comment (miri64) [reply doc/doxygen/src/mainpage.md]: Done

**Comment by emmanuelsearch** [comment]
please squash and merge when CI is green!

**Comment by miri64** [comment]
Squashed.

**Pull request merged**